### PR TITLE
Settings UI: Adds sharing tab with cards for individual sharing services.

### PR DESCRIPTION
**Pull request by dereksmart**
Fixes: #6745

This adds a new route/tab for `/sharing`

screen shot 2017-03-24 at 4 02 52 pm

Users who are linked should be taken to wordpress.com/sharing settings area, where unlinked users should see cards asking them to connect their user account.

### CODE REVIEW

**Comment by rickybanister** [comment]
Can the tab be between writing and discussion?

Otherwise looks awesome.

**Comment by dereksmart** [comment]
Moved the tab, and also fixed so that editors will have access to the publicize card.

**Comment by eliorivero** [comment]
/wp-content/plugins/jetpack/_inc/client/sharing/publicize.jsx
  49:2  error  Newline required at end of file but not found  eol-last

✖ 1 problem (1 error, 0 warnings)

Details:
    module: Module[1109]
    error: 
/wp-content/plugins/jetpack/_inc/client/sharing/publicize.jsx
  49:2  error  Newline required at end of file but not found  eol-last

✖ 1 problem (1 error, 0 warnings)

File: _inc/client/sharing/share-buttons.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+							'networks and show their support.' )
+					}
+				</SettingsGroup>
+				{
+					configCard()
+				}
+			</SettingsCard>
+		);
+	}
+}
```
Comment (eliorivero) [inline _inc/client/sharing/share-buttons.jsx]: Needs moar line breaks

Comment (dereksmart) [inline _inc/client/sharing/share-buttons.jsx]: done b93574c

**Comment by eliorivero** [comment]
Good to go!

**Pull request merged**